mkosi.images/system/mkosi.sanitizers.chroot

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -e
+
+if [[ -z "$SANITIZERS" ]]; then
+    exit 0
+fi
+
+# Sanitizers log to stderr by default. However, journald's stderr is connected to /dev/null, so we lose
+# all the sanitizer logs. To rectify that, let's connect journald's stdout to kmsg so that the sanitizer
+# failures end up in the journal.
+mkdir -p /etc/systemd/system/systemd-journald.service.d
+cat >/etc/systemd/system/systemd-journald.service.d/10-stdout-tty.conf <<EOF
+[Service]
+StandardOutput=kmsg
+EOF
+
+# ASAN and syscall filters aren't compatible with each other.
+find /usr /etc -name '*.service' -type f -exec sed -i 's/^\(MemoryDeny\|SystemCall\)/# \1/' {} +
+
+# `systemd-hwdb update` takes > 50s when built with sanitizers so let's not run it by default.
+systemctl mask systemd-hwdb-update.service
+
+ASAN_RT_PATH="$(grep libasan.so < <(ldd /usr/lib/systemd/systemd) | cut -d ' ' -f 3)"
+if [[ -z "$ASAN_RT_PATH" ]]; then
+    ASAN_RT_PATH="$(grep libclang_rt.asan < <(ldd /usr/lib/systemd/systemd) | cut -d ' ' -f 3)"
```

Comment:
It doesn't seem to be possible to build systemd with clang using mkosi currently so this part seems to be redundant. Though in terms of testing with ASan/UBSan clang makes much more sense because it catches more things.

Reply:
@evverx Shouldn't be too hard to add clang, I'll see about adding that to this PR.

Reply:
I think it would probably be easier to introduce it in a separate PR because it comes with its quirks like `-share-libasan`, https://github.com/systemd/systemd/issues/29972 and so on.

Reply:
FWIW it would be great if it was possible to turn on the verbose mode to see what is going on. Currently it's necessary to manually delete things like `terse` in `mkosi.build.chroot` to find out that `-Werror` isn't passed anywhere for example.